lib/promscrape/discovery/ec2/instance.go

```diff
@@ -4,12 +4,19 @@ import (
 	"encoding/xml"
 	"fmt"
 	"strings"
+	"time"
 
 	"github.com/VictoriaMetrics/VictoriaMetrics/lib/promscrape/discoveryutils"
 )
 
 // getInstancesLabels returns labels for ec2 instances obtained from the given cfg
 func getInstancesLabels(cfg *apiConfig) ([]map[string]string, error) {
+	// refresh api credentials if needed
```

Comment:
I think it is better moving the check for expired role creds into `getEC2APIResponse()`